embrace-android-core/src/main/kotlin/io/embrace/android/embracesdk/internal/session/orchestrator/V2PayloadStore.kt

```diff
@@ -27,14 +30,30 @@ internal class V2PayloadStore(
     }
 
     override fun storeLogPayload(envelope: Envelope<LogPayload>, attemptImmediateRequest: Boolean) {
-        val type = SupportedEnvelopeType.LOG // TODO: distinguish between log/network/crash type
+        val type = findSupportedEnvelopeType(envelope.data.logs)
         intakeService.take(envelope, createMetadata(type))
     }
 
     override fun handleCrash(crashId: String) {
         intakeService.shutdown()
     }
 
+    private fun findSupportedEnvelopeType(logs: List<Log>?): SupportedEnvelopeType {
+        // look at emb.type in all the logs. If the type is not the same for all, default to LOG.
+        val embType = logs?.mapNotNull {
```

Comment:
If we are assuming log payloads are homogenous, I'd rather just look at the first log and its type and use that. I can see that in the future, we get OTel logs that don't have `emb.type`, and we'll just treat them generically and not specify a type in the envelope.

Reply:
Updated implementation to achieve this.